raiden/api/rest.py

```diff
@@ -257,6 +258,16 @@ def open(self, partner_address, token_address, settle_timeout, balance=None):
             )
         except (InvalidAddress, InvalidSettleTimeout, SamePeerAddress) as e:
             return make_response(str(e), httplib.CONFLICT)
+        except ValueError as e:
+            if re.match("Token address 0x[0-9a-f]{40} does not contain code", str(e)):
```

Comment:
For specific exception comparison, like these, you should prefer to create a new exception class in `raiden/exceptions.py`, and raise it instead of generic `ValueError` ones with string comparison, as it is an anti-pattern, handling it like the tuple above.